LCM/dsc/engine/ConfigurationManager/LocalConfigurationManager.c

```diff
@@ -63,11 +63,72 @@ static LCMTaskNode * g_TaskHead = NULL;
 static pthread_mutex_t g_TaskQueueMutex;
 static int g_TaskQueueShutdown = 0;
 static int g_TaskQueueExists = 0;
+static MI_Char g_CurrentRunningMethodName[LCM_MAX_PATH] = {0};
+
+MI_Boolean TaskAllowedToAddInQueue(LCMTaskNode * node)
+{
+    LCMTaskNode* current;
+    MI_Boolean result = MI_TRUE;
+    pthread_mutex_lock(&g_TaskQueueMutex);
+
+    Context_Invoke_Basic *nodeArgs = (Context_Invoke_Basic *)node->params;
+    if(nodeArgs == NULL || nodeArgs->methodName == NULL)
+    {
+        goto Cleanup;
+    }
+
+    // if incoming request is consistency request.
+    if(Tcscasecmp(nodeArgs->methodName, MSFT_DSCLocalConfigManager_PerformRequiredConfigurationChecks) == 0)
+    {
+        // verify if consistency operation is already queued.
+        current = g_TaskHead;
+        while (current != NULL)
+        {
+            Context_Invoke_Basic *currentNodeArgs = current->params;
+            if(currentNodeArgs == NULL || currentNodeArgs->methodName == NULL)
+            {
+                current = current->next;
+                continue;
+            }
+
+            // Only one consistency operation can be queued.
+            if(Tcscasecmp(currentNodeArgs->methodName, MSFT_DSCLocalConfigManager_PerformRequiredConfigurationChecks) == 0)
+            {
+                DSC_EventWriteMessageDuplicateDscOperation(currentNodeArgs->methodName);
+                result = MI_FALSE;
+                goto Cleanup;
+            }
+
+            current = current->next;
+        }
+
+        // verify if consistency operation is already running.
+        if(Tcscasecmp(g_CurrentRunningMethodName, MSFT_DSCLocalConfigManager_PerformRequiredConfigurationChecks) == 0)
+        {
+            DSC_EventWriteMessageDuplicateDscOperation(nodeArgs->methodName);
```

Comment:
As we are adding more logging, it would be good to differentiate between the 2 cases when we don't add to the queue - 1) when the same item is in the queue 2) when the same item is currently being processed

Reply:
Will do that. Thanks

---
In reply to: [138802133](https://github.com/Microsoft/PowerShell-DSC-for-Linux/pull/323#discussion_r138802133) [](ancestors = 138802133)